aminator/plugins/cloud/ec2.py

```diff
@@ -314,24 +314,34 @@ def _register_image(self, **ami_metadata):
         request['Name'] = ami_metadata.get('name', None)
         request['Description'] = ami_metadata.get('description', None)
         request['Architecture'] = ami_metadata.get('architecture', None)
-        request['BlockDeviceMappings'] = ami_metadata.get('block_device_map', None)
-        request['VirtualizationType'] = ami_metadata.get('virtualization_type', None)
-        request['RootDeviceName'] = ami_metadata.get('root_device_name', None)
         request['EnaSupport'] = ami_metadata.get('ena_networking', False)
+        request['VirtualizationType'] = ami_metadata.get('virtualization_type', None)
 
-        if (ami_metadata.get('virtualization_type') == 'pv'):
-            request['KernelId'] = ami_metadata.get('kernel_id', None)
-            request['RamdiskId'] = ami_metadata.get('ramdisk_id', None)
+        # when instance store, don't provide botocore expects a string value
+        if ami_metadata.get('block_device_map') is not None:
+            request['BlockDeviceMappings'] = ami_metadata.get('block_device_map')
+        if ami_metadata.get('root_device_name') is not None:
+            request['RootDeviceName'] = ami_metadata.get('root_device_name')
 
-        # assert we have all the key params we need, nothing should be None
-        for key, value in request.items():
-            if request[key] is None:
-                raise FinalizerException('{} cannot be None'.format(key))
+        # only present for instance store
+        if ami_metadata.get('image_location') is not None:
+            request['ImageLocation'] = ami_metadata.get('image_location')
 
         # can only be set to 'simple' for hvm.  don't include otherwise
         if ami_metadata.get('sriov_net_support') is not None:
             request['SriovNetSupport'] = ami_metadata.get('sriov_net_support')
 
+        if (ami_metadata.get('virtualization_type') == 'paravirtual'):
+            # KernelId required
+            request['KernelId'] = ami_metadata.get('kernel_id', None)
+            if ami_metadata.get('ramdisk_id') is not None:
+                request['RamdiskId'] = ami_metadata.get('ramdisk_id', None)
+
+        # assert we have all the key params. Nothing to _here_ should be None
+        for key, value in request.items():
+            if request[key] is None:
```

Comment:
You're iterating over the items and have the key and value in key and value already, so you can just test for value being none, no reason to index into the dict again